bundle/src/main/java/com/adobe/acs/commons/mcp/impl/processes/AssetFolderCreator.java

```diff
@@ -0,0 +1,376 @@
+            // Try using the fallback converter if the primary convert could not resolve to a valid definition.
+            if (assetFolderDefinition == null) {
+                assetFolderDefinition = getAssetFolderDefinition(fallback, cellValue, previousAssetFolderPath);
+            }
+
+            if (assetFolderDefinition == null) {
+                log.warn("Could not find a Asset Folder Converter that accepts value [ {} ]; skipping...", cellValue);
+                // Record parse failure
+                record(ReportRowStatus.FAILED_TO_PARSE, "", cellValue);
+                throw new IllegalArgumentException(String.format("Unable to parse value [ %s ]. Skipping rest of row to prevent undesired structured from being created.", cellValue));
+            } else {
+                /* Prepare for next Cell */
+                if (assetFolderDefinitions.get(assetFolderDefinition.getId()) == null) {
+                    assetFolderDefinitions.put(assetFolderDefinition.getId(), assetFolderDefinition);
+                }
+
+                return assetFolderDefinition.getPath();
+            }
+        } else {
+            // If cell is blank then treat as it it is empty.
+            return previousAssetFolderPath;
+        }
+    }
+
+    public void createAssetFolders(ActionManager manager) {
+        assetFolderDefinitions.values().stream().forEach(assetFolderDefinition -> {
+            try {
+                manager.withResolver(rr -> {
+                    createAssetFolder(assetFolderDefinition, rr);
+                });
+            } catch (Exception e) {
+                log.error("Unable to import asset folders via ACS Commons MCP - Asset Folder Creator", e);
+            }
+        });
+    }
+
+    /**
+     * Creates an Asset Folder.
+     *
+     * @param assetFolderDefinition the asset folder definition to create.
+     * @param resourceResolver the resource resolver object used to create the asset folder.
+     * @throws PersistenceException
+     * @throws RepositoryException
+     */
+    protected void createAssetFolder(final AssetFolderDefinition assetFolderDefinition, final ResourceResolver resourceResolver) {
+        ReportRowStatus status;
+
+        Resource folder = resourceResolver.getResource(assetFolderDefinition.getPath());
+
+        try {
+            if (folder == null) {
+                final Map<String, Object> folderProperties = new HashMap<>();
+                folderProperties.put(JcrConstants.JCR_PRIMARYTYPE, assetFolderDefinition.getNodeType());
+                folder = resourceResolver.create(resourceResolver.getResource(assetFolderDefinition.getParentPath()),
+                        assetFolderDefinition.getName(),
+                        folderProperties);
+
+                status = ReportRowStatus.CREATED;
+            } else {
+                status = ReportRowStatus.UPDATED_FOLDER_TITLES;
+            }
+
+            final Resource jcrContent = folder.getChild(JcrConstants.JCR_CONTENT);
```

Comment:
This kind of null checking makes me pine for a JCR API that embraces the new Optional stuff.  But I digress, this looks fine.